File: package/main/src/String/pascalCase.ts
import { capitalizeWord } from "./capitalizeWord";
import { words } from "./words";

/**
 * Converts a string to PascalCase.
 * @param {string} string_ - Input string
 * @returns {string} PascalCase string
 * @example
 * pascalCase("hello-world"); // "HelloWorld"
 * pascalCase("hello_world"); // "HelloWorld"
 */
export const pascalCase = (string_: string): string =>
  words(string_)
    .map((word) => capitalizeWord(word))
    .join("");

File: package/umt_python/src/string/pascal_case.py
from .capitalize_word import capitalize_word
from .words import words


def pascal_case(string_: str) -> str:
    """
    Converts a string to PascalCase.

    Args:
        string_: Input string.

    Returns:
        PascalCase string.

    Example:
        >>> pascal_case("hello-world")
        'HelloWorld'
        >>> pascal_case("hello_world")
        'HelloWorld'
        >>> pascal_case("helloWorld")
        'HelloWorld'
        >>> pascal_case("XMLHttpRequest")
        'XmlHttpRequest'
    """
    return "".join(capitalize_word(word) for word in words(string_))
